include/ncengine/ecs/Component.h

```diff
@@ -76,6 +83,31 @@ struct DefaultStoragePolicy
 template<PooledComponent T>
 struct StoragePolicy : DefaultStoragePolicy {};
 
+/** @brief Optional callbacks for generic component operations. */
+template<PooledComponent T>
+struct ComponentHandler
+{
+    /** @brief Function type for the DrawUI handler. */
+    using DrawUI_t = std::function<void(T&)>;
+
+    /** @brief Signal type for on add events. */
+    using OnAdd_t = Signal<T&>;
```

Comment:
Should this and OnRemove be optional?

Reply:
Since the decision to dispatch to onAdd/onRemove is at compile time, I had initially (not this PR, like early registry stuff) tried to make this a `std::conditional` (sort of a compile time variant), so that we could null the type if it wasn't needed, but never got it working completely.

The `std::optional` approach is a little spooky - we'd have compile time in/exclusion of code that accesses it, but runtime in/exclusion of the actual signal. We'd always have to be checking the optional state, somewhat defeating the purpose of compile time stuff.

After looking at this again, I'm thinking maybe this shouldn't have moved here at all, and it should stay in `PerComponentStorage`
1. The purpose of `ComponentHandler` was to drive `AnyComponent` behavior. The signals are a concept totally separate from `AnyComponent`.
2. As far as configuration of the signals, the user should only be be concerned with setting the `StoragePolicy` correctly, and connecting systems to the signals. There is added value in having the user construct one and pass it in.

Reply:
Moved them back to where they were